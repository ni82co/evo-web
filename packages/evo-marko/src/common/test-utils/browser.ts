import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el: HTMLElement, to: number, cb: Function) {
  fireEvent.scroll(el);
  el.scrollLeft = to;
  setTimeout(cb, 600);
}

function waitFrames(count: number, cb: Function) {
  if (count) {
    return requestAnimationFrame(() => {
      waitFrames(count - 1, cb);
    });
  }

  cb();
}

interface FastAnimations {
  fastAnimationStyle?: HTMLStyleElement;
  start: () => void;
  stop: () => void;
}

const fastAnimations:FastAnimations = {
  // Adds an style to the document which forces all transitions to run more quickly for the tests.
  start() {
    if (this.fastAnimationStyle) {
      return;
    }

    this.fastAnimationStyle = document.createElement("style");
    this.fastAnimationStyle.innerHTML = `* {
                transition-duration: 0.1s !important;
                transition-delay: 0.1s !important;
                scroll-behavior: auto !important;
            }`;
    document.head.appendChild(this.fastAnimationStyle);
  },
  stop() {
    if (!this.fastAnimationStyle) {
      return;
    }

    document.head.removeChild(this.fastAnimationStyle);
    this.fastAnimationStyle = undefined;
  },
};

export { fastAnimations, waitFrames, simulateScroll };
