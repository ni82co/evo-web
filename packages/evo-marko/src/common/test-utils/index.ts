export * from "./browser"
export * from "./snapshots";