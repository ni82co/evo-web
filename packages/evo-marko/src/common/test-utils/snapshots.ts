import { render, normalize } from "@marko/testing-library";
import { expect } from "vitest";

async function snapshotHTML(template: any, input?: any) {
  const { container } = await render(template, input);
  expect(normalize(container)).toMatchSnapshot();
}

export { snapshotHTML };
