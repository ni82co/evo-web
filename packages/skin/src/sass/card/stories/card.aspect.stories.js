export default { title: "Skin/Card/Aspect" };

export const verticalAspectSquare = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const verticalAspect_5by4 = () => `
<span class="card card--aspect-5-4">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Football rookie cards so pristine that they've never been touched by human hands, only examined through specialized gloves by card whisperers.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const verticalAspect_9by16 = () => `
<span class="card card--aspect-16-9">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine â€“ stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>

        <div class="card__footer">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;
