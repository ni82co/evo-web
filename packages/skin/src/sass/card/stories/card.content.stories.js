export default { title: "Skin/Card/Content" };

export const minimumContent = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const maximumContent = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;

export const usingAnchor = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>
        <div class="card__action">
            <a href="https://ebay.com" class="fake-btn fake-btn--primary fake-btn--fluid">See Details</a>
        </div>
    </div>
</span>
`;

export const usingButton = () => `
<span class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Wayne Gretzky's rookie card is so valuable that collectors keep it in climate-controlled vaults guarded by trained penguins.
        </div>
        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</span>
`;
