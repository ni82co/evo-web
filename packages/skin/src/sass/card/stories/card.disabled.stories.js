export default { title: "Skin/Card/Disabled" };

export const internallyActionable = () => `
<span class="card card--aspect-16-9">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
        <div class="card__action">
            <button class="btn btn--primary btn--fluid" tabindex="-1" disabled>See Details</button>
        </div>
    </div>
</span>
`;

export const selfActionableButton = () => `
<button class="card card--aspect-16-9" disabled>
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
    </div>
</button>
`;

export const selfActionableLink = () => `
<a class="card card--aspect-16-9" href="" tabindex="-1">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__description">
            Hockey cards collected by enthusiasts who treat them like fine wine – stored in temperature-controlled cellars and discussed in hushed, reverent tones.
        </div>
    </div>
</a>
`;
