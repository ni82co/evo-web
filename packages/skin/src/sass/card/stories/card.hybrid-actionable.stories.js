export default { title: "Skin/Card/Hybrid Actionable" };

export const vertical = () => `
<div class="card card--outlined">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <div class="card__title">Authentic Rookie Cards</div>
        <div class="card__description">
            Rookie cards that serve as a hedge against the apocalypse – when society collapses, surely someone will trade food for a mint Luka Dončić card.
        </div>

        <div class="card__action">
            <button class="btn btn--primary btn--fluid">See Details</button>
        </div>
    </div>
</div>
`;

export const horizontal = () => `
<div class="card card--horizontal card--outlined">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <div class="card__overline">
            <svg class="icon icon--16" height="16" width="16">
                <use href="#icon-authenticity-guarantee-filled-16-colored" />
            </svg>
            <span class="card__overline-text">Authenticity Guaranteed</span>
        </div>
        <div class="card__title">Authentic Rookie Cards</div>
        <div class="card__description">
            Rookie cards that serve as a hedge against the apocalypse – when society collapses, surely someone will trade food for a mint Luka Dončić card.
        </div>

        <div class="card__action">
            <button class="btn btn--primary">See Details</button>
        </div>
    </div>
</div>
`;
