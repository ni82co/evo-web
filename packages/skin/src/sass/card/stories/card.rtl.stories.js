export default { title: "Skin/Card/RTL" };

export const vertical = () => `
<div dir="rtl">
    <span class="card">
        <div class="card__media">
            <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
        </div>
        <div class="card__main">
            <div class="card__overline">
                <svg class="icon icon--16" height="16" width="16">
                    <use href="#icon-authenticity-guarantee-filled-16-colored" />
                </svg>
                <span class="card__overline-text">Authenticity Guaranteed</span>
            </div>
            <h3 class="card__title">Authentic Rookie Cards</h3>
            <div class="card__description">
                This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
            </div>

            <div class="card__footer">
                <button class="btn btn--primary btn--fluid">See Details</button>
            </div>
        </div>
    </span>
</div>
`;

export const horizontal = () => `
<div dir="rtl">
    <span class="card card--horizontal">
        <div class="card__media">
            <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
        </div>
        <div class="card__main">
            <div class="card__overline">
                <svg class="icon icon--16" height="16" width="16">
                    <use href="#icon-authenticity-guarantee-filled-16-colored" />
                </svg>
                <span class="card__overline-text">Authenticity Guaranteed</span>
            </div>
            <h3 class="card__title">Authentic Rookie Cards</h3>
            <div class="card__description">
                This legendary rookie card collection features future Hall of Famers back when their biggest concern was remembering to smile for the camera.
            </div>

            <div class="card__footer">
                <button class="btn btn--primary">See Details</button>
            </div>
        </div>
    </span>
</div>
`;