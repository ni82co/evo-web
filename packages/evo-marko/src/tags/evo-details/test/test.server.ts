import { describe, it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../details.stories"; // import all stories from the stories file
const { Default } = composeStories(stories);

describe("details", () => {
  it("renders basic version", async () => {
    await snapshotHTML(Default);
  });

  it("renders as div version", async () => {
    await snapshotHTML(Default, {as: "div"});
  });

  it("renders in open state", async () => {
    await snapshotHTML(Default, {open: true});
  });

  it("renders small version", async () => {
    await snapshotHTML(Default, {size: "small"});
  });

  it("renders center version", async () => {
    await snapshotHTML(Default, {alignment: "center" });
  });
});
