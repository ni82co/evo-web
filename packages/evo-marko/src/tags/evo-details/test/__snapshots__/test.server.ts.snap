// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`details > renders as div version 1`] = `
<DocumentFragment>
  <details
    class="details"
  >
    <summary
      class="details__summary"
    >
      <span
        class="details__label"
      >
        Details
      </span>
      <span
        class="details__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M8.707 12.707a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L8 10.586l5.293-5.293a1 1 0 1 1 1.414 1.414l-6 6Z"
            />
          </symbol>
          <use
            href="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </summary>
    <div
      class="details__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </div>
  </details>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.b=[0,1,{content:_._["src/tags/evo-details/examples/default.marko_1_renderer"](_.a={}),htmlInput:{}},3,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",6,"src/tags/evo-details/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`details > renders basic version 1`] = `
<DocumentFragment>
  <details
    class="details"
  >
    <summary
      class="details__summary"
    >
      <span
        class="details__label"
      >
        Details
      </span>
      <span
        class="details__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M8.707 12.707a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L8 10.586l5.293-5.293a1 1 0 1 1 1.414 1.414l-6 6Z"
            />
          </symbol>
          <use
            href="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </summary>
    <div
      class="details__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </div>
  </details>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.b=[0,1,{content:_._["src/tags/evo-details/examples/default.marko_1_renderer"](_.a={}),htmlInput:{}},3,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",6,"src/tags/evo-details/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`details > renders center version 1`] = `
<DocumentFragment>
  <details
    class="details"
  >
    <summary
      class="details__summary details__summary--center"
    >
      <span
        class="details__label"
      >
        Details
      </span>
      <span
        class="details__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M8.707 12.707a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L8 10.586l5.293-5.293a1 1 0 1 1 1.414 1.414l-6 6Z"
            />
          </symbol>
          <use
            href="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </summary>
    <div
      class="details__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </div>
  </details>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.b=[0,1,{content:_._["src/tags/evo-details/examples/default.marko_1_renderer"](_.a={}),htmlInput:{}},3,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",6,"src/tags/evo-details/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`details > renders in open state 1`] = `
<DocumentFragment>
  <details
    class="details"
    open=""
  >
    <summary
      class="details__summary"
    >
      <span
        class="details__label"
      >
        Details
      </span>
      <span
        class="details__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M8.707 12.707a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L8 10.586l5.293-5.293a1 1 0 1 1 1.414 1.414l-6 6Z"
            />
          </symbol>
          <use
            href="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </summary>
    <div
      class="details__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </div>
  </details>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.b=[0,1,{content:_._["src/tags/evo-details/examples/default.marko_1_renderer"](_.a={}),htmlInput:{open:!0}},3,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",6,"src/tags/evo-details/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`details > renders small version 1`] = `
<DocumentFragment>
  <details
    class="details"
  >
    <summary
      class="details__summary details__summary--small"
    >
      <span
        class="details__label"
      >
        Details
      </span>
      <span
        class="details__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M8.707 12.707a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L8 10.586l5.293-5.293a1 1 0 1 1 1.414 1.414l-6 6Z"
            />
          </symbol>
          <use
            href="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </summary>
    <div
      class="details__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </div>
  </details>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.b=[0,1,{content:_._["src/tags/evo-details/examples/default.marko_1_renderer"](_.a={}),htmlInput:{}},3,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",6,"src/tags/evo-details/index.marko_0_htmlInput",2];M._.w()
  </script>
</DocumentFragment>
`;
